src/dbnode/storage/repair.go

```diff
@@ -441,3 +480,194 @@ func (r repairerNoOp) Start()        {}
+		if err != nil {
+			return err
+		}
+		defer peerSeriesIter.Close()
+
+		tmpCtx.Reset()
+		defer tmpCtx.BlockingClose()
+		localSeriesDataBlocks, err := shard.ReadEncoded(tmpCtx, seriesID, start, end, nsCtx)
+		if err != nil {
+			return err
+		}
+		localSeriesSliceOfSlices := xio.NewReaderSliceOfSlicesFromBlockReadersIterator(localSeriesDataBlocks)
+		localSeriesIter := r.opts.MultiReaderIteratorPool().Get()
+		localSeriesIter.ResetSliceOfSlices(localSeriesSliceOfSlices, nsCtx.Schema)
+
+		var (
+			i             = 0
+			foundMismatch = false
+		)
+		for localSeriesIter.Next() {
+			if !peerSeriesIter.Next() {
+				r.logger.Error(
+					"series had next locally, but not from peers",
+					zap.String("namespace", nsCtx.ID.String()),
+					zap.Time("start", start),
+					zap.Time("end", end),
+					zap.String("series", seriesID.String()),
+					zap.Error(peerSeriesIter.Err()),
+				)
+				foundMismatch = true
+				break
+			}
+
+			localDP, localUnit, localAnnotation := localSeriesIter.Current()
+			peerDP, peerUnit, peerAnnotation := peerSeriesIter.Current()
+
+			if !localDP.Timestamp.Equal(peerDP.Timestamp) {
+				r.logger.Error(
+					"timestamps did not match",
+					zap.Int("index", i),
+					zap.Time("local", localDP.Timestamp),
+					zap.Time("peer", peerDP.Timestamp),
+				)
+				foundMismatch = true
+				break
+			}
+
+			if localDP.Value != peerDP.Value {
+				r.logger.Error(
+					"Values did not match",
+					zap.Int("index", i),
+					zap.Float64("local", localDP.Value),
+					zap.Float64("peer", peerDP.Value),
+				)
+				foundMismatch = true
+				break
+			}
+
+			if localUnit != peerUnit {
+				r.logger.Error(
+					"Values did not match",
+					zap.Int("index", i),
+					zap.Int("local", int(localUnit)),
+					zap.Int("peer", int(peerUnit)),
+				)
+				foundMismatch = true
+				break
+			}
+
+			if nsCtx.Schema == nil {
+				// Remaining shadow logic is proto-specific.
+				continue
+			}
+
+			err = localM.Unmarshal(localAnnotation)
+			if err != nil {
+				r.logger.Error(
+					"Unable to unmarshal local annotation",
+					zap.Int("index", i),
+					zap.Error(err),
+				)
+				foundMismatch = true
+				break
+			}
+
+			err = peerM.Unmarshal(peerAnnotation)
+			if err != nil {
+				r.logger.Error(
+					"Unable to unmarshal peer annotation",
+					zap.Int("index", i),
+					zap.Error(err),
+				)
+				foundMismatch = true
+				break
+			}
+
+			if !dynamic.Equal(localM, peerM) {
+				r.logger.Error(
+					"Local message does not equal peer message",
+					zap.Int("index", i),
+					zap.String("local", localM.String()),
+					zap.String("peer", peerM.String()),
+				)
+				foundMismatch = true
+				break
+			}
+
+			if !bytes.Equal(localAnnotation, peerAnnotation) {
+				r.logger.Error(
+					"Local message equals peer message, but annotations do not match",
+					zap.Int("index", i),
+					zap.String("local", string(localAnnotation)),
+					zap.String("peer", string(peerAnnotation)),
```

Comment:
Wondering if there's value in logging the `string`s of these annotations. Would it be readable in any way?

Reply:
might end up being useful for deep debugging, better than having to add it yourself later

Reply:
Sure.